integrated_channels/utils.py

```diff
@@ -161,20 +161,26 @@ def get_image_url(content_metadata_item):
     return image_url
 
 
-def is_already_transmitted(transmission, enterprise_enrollment_id, grade):
+def is_already_transmitted(transmission, enterprise_enrollment_id, grade, subsection_id=None):
     """
     Returns: Boolean indicating if completion date for given enrollment is already sent of not.
 
     Args:
         transmission: TransmissionAudit model to search enrollment in
         enterprise_enrollment_id: enrollment id
         grade: 'Pass' or 'Fail' status
+        subsection_id (Optional): The id of the subsection, needed if there exist multiple audits per course
     """
     try:
         already_transmitted = transmission.objects.filter(
             enterprise_course_enrollment_id=enterprise_enrollment_id,
             error_message='',
-        ).latest('id')
+        )
+        if subsection_id:
+            already_transmitted.filter(subsection_id=subsection_id)
+
+        already_transmitted = already_transmitted.latest('id')
```

Comment:
So we see if a course exists in the transmission logs for grades and then if we're trying to actually check assessments, we then filter the already-transmitted course down to subsection, but you have `already_transmitted.filter(subsection_id=subsection_id)` without assignment so does it still save the filtered result set when you then call `already_transmitted.latest('id')

Reply:
I was working under the assumption that the `alread_transmitted` object would retain the filter. I believe we do something similar in the base course level exporter ie 
```
        enrollment_queryset = EnterpriseCourseEnrollment.objects.select_related(
            'enterprise_customer_user'
        ).filter(
            enterprise_customer_user__enterprise_customer=self.enterprise_customer,
            enterprise_customer_user__active=True,
        )
        if learner_to_transmit and course_run_id:
            enrollment_queryset = enrollment_queryset.filter(
                course_id=course_run_id,
                enterprise_customer_user__user_id=learner_to_transmit.id,
            )
            exporting_single_learner = True
            generate_formatted_log(
                'Exporting single learner. Course: {course_run}, User: {user_id}'.format(
                    course_run=course_run_id,
                    user_id=learner_to_transmit.id
                ),
                channel_name=channel_name,
                enterprise_customer_identifier=self.enterprise_customer.name
            )
        enrollment_queryset = enrollment_queryset.order_by('course_id')
```
But perhaps this is slightly different? I would assume that the `.latest('id)` is still necessary for the assessment level transmission because we may have multiple records for a single particular subsection and need to get the latest one. Please correct me if I'm wrong here!

Reply:
yeah but where we're filtering in that example we're setting assignment:
```
enrollment_queryset = enrollment_queryset.filter(
                course_id=course_run_id,
                enterprise_customer_user__user_id=learner_to_transmit.id,
            )
```
but yours is just:
```
        if subsection_id:
            already_transmitted.filter(subsection_id=subsection_id)
```
That's what I'm confused by.

Reply:
I'm a bit doubtful running a filter on an object inherently updates/saves the object with the filter without assignment? But I could be wrong.

Reply:
hmm did not think filter did any updates... https://docs.djangoproject.com/en/3.1/ref/models/querysets/#filter you may have to assign the output of filter back if further procesing

Reply:
Talia and I labbed this and are (I hope) happy with how it's handled.